cmd/gardenlet/app/app.go

```diff
@@ -525,6 +578,44 @@ func init() {
 	seedManagementDecoder = serializer.NewCodecFactory(scheme).UniversalDeserializer()
 }
 
+// readDefaultDomainSecrets reads and sorts default domain secrets by priority
+func (g *garden) readDefaultDomainSecrets(ctx context.Context, gardenClient client.Client) ([]corev1.Secret, error) {
```

Comment:
I see two more implementations of reading the global default domain secrets just in this PR, can you extract this into a function and re-use it? 
The other occcurances are in 
- `pkg/utils/gardener.ReadGardenDefaultDomains`
- `plugin/pkg/utils.getGlobalDefaultDomains`

There are 2 more legacy implementation doing similar job.